keyboards/rart/rartlite/rartlite.c

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2021 Alabahuy
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rartlite.h"
+
+void keyboard_pre_init_user(void) {
+  // Call the keyboard pre init code.
+  // Set our LED pins as output
+  setPinOutput(F5);
+  setPinOutput(F6);
+  setPinOutput(B1);
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(F5, led_state.num_lock);
+        writePin(F6, led_state.caps_lock);
+    }
+    return res;
+}
+
+layer_state_t layer_state_set_kb(layer_state_t state) {
+    switch (get_highest_layer(state)) {
+      case 1:
+        writePinHigh(B1);
+        break;
+      default: //  for any other layers, or the default layer
+        writePinLow(B1);
+        break;
+      }
+    return layer_state_set_user(state);
+}
```

Comment:
```suggestion
void keyboard_pre_init_user(void) {
  setPinOutput(B1);
}

layer_state_t layer_state_set_kb(layer_state_t state) {
    writePin(B1, layer_state_cmp(state, 1));
    return layer_state_set_user(state);
}
```

The rest can be replaced with the following in your config.h:
```c
#define LED_NUM_LOCK_PIN F5
#define LED_NUM_LOCK_PIN F6
```

And you may or may not need `#define LED_PIN_ON_STATE 0`